From c30864eb6d35bbaa4fe57cadc5fe73cd149abb16 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Sat, 12 Apr 2025 01:32:45 +0200
Subject: [PATCH] Fix creating main window

The window property is used by conditions in multiple places, but it was
always available due to the getter function, which is not the expected
behavior. Turn it into an empty property, and assign the window only in
the open() function, which is the only place where the window is supposed
to be created.

This fixes the problem that set_application() has no effect, because it's
called after the window is presented by _on_file_manager_state_changed().
---
 gaphor/ui/mainwindow.py | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/gaphor/ui/mainwindow.py b/gaphor/ui/mainwindow.py
index 6300037be..083b83a1a 100644
--- a/gaphor/ui/mainwindow.py
+++ b/gaphor/ui/mainwindow.py
@@ -98,6 +98,7 @@ class MainWindow(Service, ActionProvider):
         self.tools_menu = tools_menu
 
         self._builder: Gtk.Builder | None = new_builder()
+        self.window: Gtk.Window | None = None
         self.action_group: Gio.ActionGroup = None
         self.modeling_language_name = None
         self.diagram_types = None
@@ -129,10 +130,6 @@ class MainWindow(Service, ActionProvider):
         em.unsubscribe(self._on_modeling_language_selection_changed)
         em.unsubscribe(self._on_notification)
 
-    @property
-    def window(self):
-        return self._builder.get_object("main-window") if self._builder else None
-
     @property
     def title(self):
         return self._builder.get_object("title") if self._builder else None
@@ -167,6 +164,10 @@ class MainWindow(Service, ActionProvider):
 
         builder = self._builder
         assert builder
+
+        if not self.window:
+            self.window = builder.get_object("main-window")
+
         window = self.window
         window.set_application(gtk_app)
         if ".dev" in distribution().version:
-- 
2.49.0

